Sources/Kommunicate/Classes/KMConversationViewController.swift

```diff
@@ -178,7 +182,51 @@ open class KMConversationViewController: ALKConversationViewController {
         })
     }
 
-    @objc override open func pushNotification(notification: NSNotification) {
+    open override func addMessagesToList(_ messageList: [Any]) {
+        guard let messages = messageList as? [ALMessage] else { return }
+        messageArray = messages
+        count = 0
+        timer = Timer.scheduledTimer(timeInterval: 3, target: self, selector: #selector(loopOverMessageArray), userInfo: nil, repeats: true)
+        timer.fire()
+    }
+    
+    @objc func loopOverMessageArray() {
+        
+        if count >= messageArray.count {
+            timer.invalidate()
+            return
+        }
+        var filteredArray = [ALMessage]()
+        let contactService = ALContactService()
+        if viewModel.channelKey != nil, viewModel.channelKey == messageArray[count].groupId {
+            let delayInterval = KMAppUserDefaultHandler.shared.botMessageDelayInterval
+            UserDefaults.standard.set((delayInterval/1000), forKey: "botDelayInterval")
+            let alContact = contactService.loadContact(byKey: "userId", value:  messageArray[count].to)
+            if delayInterval > 0 && alContact?.roleType == NSNumber.init(value: AL_BOT.rawValue){
+                showTypingLabel(status: true, userId: messageArray[count].to)
+                currentMessage = messageArray[count]
+                Timer.scheduledTimer(timeInterval: TimeInterval(UserDefaults.standard.integer(forKey: "botDelayInterval")*2), target: self, selector: #selector(self.addMessagesToViewModel), userInfo: nil, repeats: false)
```

Comment:
This UserDefaults value can return nil, so put it in a separate variable and add a check.